File: editor/en/[1165]Single-Row Keyboard.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int calculateTime(String keyboard, String word) {
        final int LEN = 26;
        int[] arr = new int[LEN];

        for (int i = 0; i < LEN; i++) {
            arr[keyboard.charAt(i) - 'a'] = i;
        }

        int time = 0;
        int prevIndex = 0;

        for (char c : word.toCharArray()) {
            int currIndex = arr[c - 'a'];
            time += Math.abs(currIndex - prevIndex);
            prevIndex = currIndex;
        }

        return time;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1165]Single-Row Keyboard.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} keyboard
 * @param {string} word
 * @return {number}
 */
var calculateTime = function (keyboard, word) {
    let charToIndexMap = new Map();

    for (let i = 0; i < keyboard.length; i++) {
        charToIndexMap.set(keyboard[i], i);
    }

    let time = 0;
    let prevIndex = 0;

    for (let c of word) {
        const currIndex = charToIndexMap.get(c);
        time += Math.abs(currIndex - prevIndex);
        prevIndex = currIndex;
    }

    return time;
};
//leetcode submit region end(Prohibit modification and deletion)
